generate/generate.go

```diff
@@ -416,6 +416,17 @@ func (g *Generator) SetLinuxResourcesCPUMems(mems string) {
 	g.spec.Linux.Resources.CPU.Mems = &mems
 }
 
+// SetLinuxResourcesHugepageLimit sets g.spec.Linux.Resources.HugepageLimits.
+func (g *Generator) SetLinuxResourcesHugepageLimit(pageSize string, limit uint64) {
```

Comment:
You probably want `Add*` or `Append*` instead of `Set*` here.  To me, `Set*` is for when you're clobbering any preexisting value.